### ISSUE TITLE
escaping new line in tables

### ISSUE THREAD

**Issue by akostadinov** [cucumber/gherkin3#146]
Hello,

I'm thinking I'm doing something wrong as I see nobody else complain but I can't see `\n` translated to a new line in tables (neither examples table nor a step table). According to README `\n` should be translated to a new line whenever met. See:
```
  @pry_outline
  Scenario Outline: I want to pry an outline
    When I pry
    Examples: second test case
      |fff|ddd|
      |aaa|b\nbb|
```
```
$ cucumber features/test/debug.feature:24
From: /home/avalon/v3/features/step_definitions/debug.rb @ line 3 :

    1: When /^I pry$/ do
    2:   require 'pry'
 => 3:   binding.pry
    4: end
    5: 
    6: When /^I pry in a step with table$/ do |table|
    7:   require 'pry'
    8:   binding.pry

[1] pry(#<CucuShift::DefaultWorld>)> cd scenario.source.last
[2] pry(#<Cucumber::Core::Ast::ExamplesTable::Row>):1> @data.values
=> ["aaa", "b\\nbb"]
```

So it appears to me as a literal `\n`. Same with step tables.
```
$ cucumber --version
2.1.0
gherkin3 (3.1.2)
```

**Comment by aslakhellesoy** [comment]
This should have been fixed in #71 which went into the 3.1.1 release. Maybe there is a bug not caught by tests? I'll have a look.

**Comment by aslakhellesoy** [comment]
I've only had a cursory look, but it seems to me gherkin3 is correctly unescaping `\n` into a newline character.

I've reproduced your problem though, so my theory is that cucumber somewhere converts the newline back to `\n` (backslash-n) somewhere?

@tooky @mattwynne @brasmusson any idea?

**Comment by brasmusson** [comment]
The [`gherkin_line.rb`](https://github.com/cucumber/gherkin3/blob/master/ruby/lib/gherkin3/gherkin_line.rb#L69) uses `'\n'` instead of `"\n"`, so no the ruby-version of gherkin3 does not correctly unescaping `\n` into a newline character. #114 has a fix for this (when making the changes to allow regexp in table cells).

**Comment by aslakhellesoy** [comment]
This is fixed on master

**Comment by akostadinov** [comment]
Can we have an official gherkin3 ruby gem release now this and other syntax related issues are resolved? I'd even version it 3.2.0 as changes are pretty serious.
